src/script/microbit-interfacing/MicrobitSerial.ts

```diff
@@ -4,82 +4,64 @@
  * SPDX-License-Identifier: MIT
  */
 
-import InputBehaviour from '../connection-behaviours/InputBehaviour';
-import MBSpecs from './MBSpecs';
-import { MicrobitConnection } from './MicrobitConnection';
+import { DeviceRequestStates } from '../stores/connectDialogStore';
+import MicrobitConnection from './MicrobitConnection';
 import MicrobitUSB from './MicrobitUSB';
+import Microbits from './Microbits';
+import { onAccelerometerChange, onButtonChange } from './change-listeners';
 import * as protocol from './serialProtocol';
+import {
+  stateOnAssigned,
```

Comment:
Just FTR, we've switched to more direct state updates here to get a clearer view of what's going on but reintroducing an interface like before isn't unreasonable.

Reply:
This would likely improve the code cycle situation.